client/client.go

```diff
@@ -58,31 +60,27 @@ func (kc *KalaClient) decode(body io.Reader, target interface{}) error {
 }
 
 func (kc *KalaClient) url(parts ...string) string {
-	for i, part := range parts {
-		if part == "" {
-			continue
-		}
-		if init, last := part[:len(part)-1], part[len(part)-1]; last == '/' {
-			parts[i] = init
-		}
-	}
 	return strings.Join(append([]string{kc.apiEndpoint}, parts...), "/") + "/"
```

Comment:
`strings.Join` will have the problem that you solved earlier with the logic you deleted here, but if you use `path.Join` rather than `strings.Join` then this should do what you want.